### ISSUE TITLE
Suggestion Add ability to autoload on the move and quit filling when full.

### ISSUE THREAD

**Issue by GWgameing** [loki79uk/FS22_UniversalAutoload#178]
This mod made it simple to add autoloading to a logging trailer I made.
I have noticed two things that would give me the possibility to configure the mod to my preferences.

It would be nice if we had the option to add something like ```<options enableLoadingOnTheMove="true"/>``` to the mod xml to allow the trailer to be loaded when the vehicle is in motion.

I would also like to have the option to have the trailer load until full. This would complement well the revision mentioned above. It would also allow for productions to be unloaded without as much user input. Again something like 
```<options enableLoadingUntilFull="true"/>``` 

I have taken a look in the UniversalAutoload.lua, but am unable to make these revisions for a pull request. I am interested to think what others in the community might have to think about this too. Thank you for your consideration.

**Comment by loki79uk** [comment]
The decision to prevent loading while moving was a workaround for limitations in the game engine, not a design preference.

If you want the details:
There is no in game function or property that can instantly impart an acceleration to an object.  So when autoloading an object we can give it a position on top of a trailer and also a velocity to match, but if the trailer is accelerating then we have to rely on "friction" between the two to increase the acceleration of the object.  In most situations the pallets will slide around and then do not end up in the intended positions.

This effect can be avoided as I do it with bale collection mode where the added bales are not added to physics, and so they do not slide around, but then you are able to pass other objects inside them, and when finally returning all objects to physics you get the kinds of bale explosions (launching into space) that we saw in FS19.  You also cannot handle any of the loaded objects (and can walk inside them) until they are returned to physics - as it was with autoloading vehicles in previous FS versions.

So basically it is not much work to allow loading while moving, but the results tend to be undesirable.

If you want to test it for yourself, simply edit this function:
```lua
function UniversalAutoload:ualGetIsMoving()
	--return self.lastSpeedReal > 0.0005
	return false
end
```

**Comment by sqncer** [comment]
I also agree option to load until full would be great. I just started using Universal Autoload while I used object storage mod from GtX for some time. I spawn pallets (9 is max), autoload them, spawn again and then there is no option to continue loading. I need to leave then enter again the vehicle, or close and open curtains in Krone trailer is also working. Or leave the spot and then return. After some time this is quite annoying. Is there some better workaround for this?

**Comment by loki79uk** [comment]
> I also agree option to load until full would be great. I just started using Universal Autoload while I used object storage mod from GtX for some time. I spawn pallets (9 is max), autoload them, spawn again and then there is no option to continue loading. I need to leave then enter again the vehicle, or close and open curtains in Krone trailer is also working. Or leave the spot and then return. After some time this is quite annoying. Is there some better workaround for this?

I don't think this is possible without recreating the bale collection mode for pallets.  I added the bale collection mode for fieldwork/courseplay, and was planning something like that with pallets for autodrive, but actually autodrive works quite nicely with just the load/unload commands.

I may add a pallet collection mode eventually, but I think that manual loading of logs is more important to add first.

**Comment by loki79uk** [comment]
> I also agree option to load until full would be great. I just started using Universal Autoload while I used object storage mod from GtX for some time. I spawn pallets (9 is max), autoload them, spawn again and then there is no option to continue loading. I need to leave then enter again the vehicle, or close and open curtains in Krone trailer is also working. Or leave the spot and then return. After some time this is quite annoying. Is there some better workaround for this?

It sounds like because the objects spawn inside the trigger then they are never detected as entering.  If that is the case, then it should happen for productions too - I will look into it.

**Comment by sqncer** [comment]
I just tried this with vanilla productions, same thing happens. Both when I spawn pallets manually or set output to spawning (not sure if this is vanilla functionality, I'm using production revamp)

**Comment by loki79uk** [comment]
Is this still happening for you?

I tried this with the GIANTS storage, and it seems to detect them as soon as they spawn:
![image](https://github.com/loki79uk/FS22_UniversalAutoload/assets/76516343/23cc5d4e-1c5a-49a7-9641-8e63fc184164)
![image](https://github.com/loki79uk/FS22_UniversalAutoload/assets/76516343/5284490b-91c4-4332-9a83-7e3e02e2817b)
